pkg/csi/controller.go

```diff
@@ -2,53 +2,148 @@ package csi
 
 import (
 	"context"
+	"fmt"
+	"math"
+
+	"github.com/oneconcern/datamon/pkg/model"
+
+	"go.uber.org/zap"
+
+	"github.com/oneconcern/datamon/pkg/core"
 
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 
 	"github.com/container-storage-interface/spec/lib/go/csi/v0"
 )
 
-type controllerServer struct { //nolint:unused
+type controllerServer struct {
+	config *controllerServerConfig
 }
 
-type controllerServerConfig struct { //nolint:unused
+type controllerServerConfig struct {
+	driver *Driver
 }
 
-func newControllerServer(config *controllerServerConfig) csi.ControllerServer { // nolint:deadcode,unparam,unused
-	return &controllerServer{}
+func newControllerServer(config *controllerServerConfig) csi.ControllerServer {
+	return &controllerServer{
+		config: config,
+	}
 }
 
 func (s *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest) (*csi.CreateVolumeResponse, error) {
-	return nil, nil
+	s.config.driver.config.Logger.Info("received create volume",
+		zap.String("name", req.Name))
+	// Check if volume==repo is there
+	repo := req.GetParameters()["repo"]
+	if repo == "" {
+		s.config.driver.config.Logger.Error("Repo not set", zap.String("name", req.Name))
+		return nil, fmt.Errorf("repo not set for req name:%s", req.Name)
+	}
+	hash := req.GetParameters()["release"]
+	if hash == "" {
+		s.config.driver.config.Logger.Error("Release not set", zap.String("name", req.Name))
+		return nil, fmt.Errorf("release not set for req name:%s", req.Name)
+	}
+	rd, err := core.GetRepo(repo, s.config.driver.metadataStore)
+	if err != nil {
+		s.config.driver.config.Logger.Error("requested repo could not be found", zap.Error(err))
+		return nil, fmt.Errorf("requested volume not found. err:" + err.Error())
+	}
+	attrs := make(map[string]string)
```

Comment:
it's possible to initialize maps with literals as in

```
attrs = map[string]string{"repo": repo, "hash": hash}
```

with some slight advantages for readability (all init-like stuff in one statement) and maintainability (not necessary to repeat the name of the map so many times)

Reply:
Yup much cleaner. Thanks